test/Microsoft.AspNet.Mvc.Core.Test/ActionResults/CreatedAtRouteResultTests.cs

```diff
@@ -0,0 +1,141 @@
+
+        [Fact]
+        public async Task CreatedAtRouteResult_ThrowsOnNullUrl()
+        {
+            // Arrange
+            var appRoot = "";
+            var response = GetMockedHttpResponseObject();
+            var httpContext = GetHttpContext(appRoot, response);
+            var actionContext = GetActionContext(httpContext);
+            var urlHelper = GetMockUrlHelper(returnValue: null);
+
+            var result = new CreatedAtRouteResult(null, new Dictionary<string, object>(), null);
+            result.UrlHelper = urlHelper;
+
+            // Act & Assert
+            await ExceptionAssert.ThrowsAsync<InvalidOperationException>(
+                async () => await result.ExecuteResultAsync(actionContext),
+            "No route matches the supplied values.");
+        }
+
+        private static HttpResponse GetMockedHttpResponseObject()
+        {
+            var stream = new MemoryStream();
+            var httpResponse = new Mock<HttpResponse>();
+            httpResponse.SetupProperty(o => o.StatusCode);
+            httpResponse.Setup(o => o.Headers).Returns(
+                new HeaderDictionary(new Dictionary<string, string[]>()));
+            httpResponse.SetupGet(o => o.Body).Returns(stream);
+            return httpResponse.Object;
+        }
+
+        private static ActionContext GetActionContext(HttpContext httpContext)
+        {
+            var routeData = new RouteData();
+            routeData.Routers.Add(new Mock<IRouter>().Object);
+
+            return new ActionContext(httpContext,
+                                    routeData,
+                                    new ActionDescriptor());
+        }
+
+        private static HttpContext GetHttpContext(string appRoot, HttpResponse response)
+        {
+            var httpContext = new Mock<HttpContext>();
+            var actionContext = GetActionContext(httpContext.Object);
+            var mockContentAccessor = new Mock<IContextAccessor<ActionContext>>();
+            mockContentAccessor.SetupGet(o => o.Value).Returns(actionContext);
+            var mockActionSelector = new Mock<IActionSelector>();
+
+            httpContext.Setup(o => o.Response)
+                       .Returns(response);
+            httpContext.Setup(o => o.RequestServices.GetService(typeof(IOutputFormattersProvider)))
+                       .Returns(new TestOutputFormatterProvider());
+            httpContext.Setup(o => o.Request.PathBase)
+                       .Returns(new PathString(appRoot));
+
+            return httpContext.Object;
+        }
+
+        private static IUrlHelper GetMockUrlHelper(string returnValue)
+        {
+            var urlHelper = new Mock<IUrlHelper>();
+            urlHelper.Setup(o => o.RouteUrl(It.IsAny<string>(), It.IsAny<object>(), It.IsAny<string>(),
+                It.IsAny<string>(), It.IsAny<string>())).Returns(returnValue);
+
+            return urlHelper.Object;
+        }
+
+        public static IEnumerable<object[]> CreatedAtRouteData
```

Comment:
The general pattern is to place `MemberData` property right above the test. Makes it easier to look up the values being used.